atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,483 @@
+
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet);
+        });
+    }
+
+    private Set<TableReference> listAllTablesWithMetadata() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT table_name from " + AtlasDbConstants.DEFAULT_ORACLE_METADATA_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet).stream()
+                    .map(TableReference::createUnsafe)
+                    .collect(Collectors.toSet());
+        });
+    }
+
+    private Set<String> listAllTablesWithMapping() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement =
+                    connection.prepareStatement("SELECT table_name from " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet);
+        });
+    }
+
+    private <T> T runWithConnection(FunctionCheckedException<Connection, T, SQLException> task) {
+        try (Connection connection = connectionManager.getConnection()) {
+            return task.apply(connection);
+        } catch (SQLException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e);
+        }
+    }
+
+    private static Set<String> getTableNamesFromResultSet(ResultSet resultSet) throws SQLException {
+        ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        while (resultSet.next()) {
+            builder.add(resultSet.getString("table_name"));
+        }
+        return builder.build();
+    }
+
+    private static String withEscapedUnderscores(String s) {
+        return s.replace("_", "\\_");
+    }
+
+    private static String withWildcardSuffix(String s) {
+        return s + "%";
+    }
+
+    private static void assertSnapshotContainsNewTableReferences(
+            AllTableDetailsSnapshot snapshot, int expectedNumberOfNewTableReferences) {
+        assertThat(snapshot.prefixedTableNames()).hasSize(expectedNumberOfNewTableReferences * 2);
+        assertThat(snapshot.physicalTableNames()).hasSize(expectedNumberOfNewTableReferences * 2);
+        assertThat(snapshot.tableReferences()).hasSize(expectedNumberOfNewTableReferences);
+    }
+
+    private void assertSnapshotIsEmptyExcludingTimestampTable(AllTableDetailsSnapshot snapshot) {
+        assertThat(snapshot.prefixedTableNames())
+                .hasSameSizeAs(snapshot.tableReferences())
+                .isEmpty();
+        if (snapshot.physicalTableNames().size() == 1) {
+            assertThat(snapshot.physicalTableNames().stream()
+                            .map(s -> s.toLowerCase(Locale.ROOT))
+                            .collect(Collectors.toSet()))
+                    .containsOnly(timestampTableName);
```

Comment:
nit: The stream's a bit unnecessary: you can just do `            assertThat(Iterables.getOnlyElement(snapshot.physicalTableNames()).toLowerCase(Locale.ROOT)).isEqualTo(timestampTableName);`.